### DISCUSSION TITLE
Replacing # with ? at the start of authorize callback query string

### DISCUSSION THREAD

**Discussion by blimpmason** [nextauthjs/next-auth#6422] [Help]
### Question 💬

Hi, I'm setting up a custom provider to work with [Mindbody](https://developers.mindbodyonline.com/PublicDocumentation/V6#oauth). 

I'm trying to find a workaround for an issue on the Mindbody side: on the authorize callback, Mindbody uses the hash (#) symbol instead of the question mark (?) at the start of the query string, so next-auth doesn't recognize the parameters passed back in the url. Here's an example callback URL (with truncated id_token):

```https://localhost:3001/api/auth/callback/mindbody#code=EC6942C60F2DBB987AAEEB7814A1C2E2AFAAB0316D1A9DE6D38E5F2B144D7766-1&id_token=eyJhbGciOiJSUzI1NiIsImtpZCI6IjlERkRDNz...&scope=email%20profile%20openid%20offline_access%20Mindbody.Api.Public.v6&session_state=_wkbVT58IMOw6qvMTjaK0Oi7Du2qdjFw3rtxuz7j7fs.35ACFD307C77A82A51BEACBF03C685A4```

I get an OAuth callback error every time unless I manually replace the # with a ? in which case the login succeeds.

Here's what my provider object looks like:
```
{
	id: "mindbody",
	name: "Mindbody",
	type: "oauth",
	wellKnown: "https://signin.mindbodyonline.com/.well-known/openid-configuration",
	clientId: "2a25eb7d-cae6-44d9-a90a-3aea3e5d77d5",
	clientSecret: "a1uasF9kV6axVU1p7Lf3qpNzkkaui33HxGixc1lT/Qc=",
	checks: ["nonce"],
	authorization: {
		params: {
			redirect_uri: "https://localhost:3001/api/auth/callback/mindbody",
			response_type: "code id_token",
			scope: "email profile openid offline_access Mindbody.Api.Public.v6",
		},
	},
	token: {
		params: {
			grant_type: "authorization_code",
		},
	},
	profile(profile) {
		return {
			id: "testId",
		};
	},
},
```

I'm experimenting with a redirect callback to replace the # but I only ever get the base URL:
```
callbacks: {
	async redirect({ url, baseUrl }) {
		console.log(url);
		return url.replace("mindbody#", "mindbody?");
	},
},
```

### How to reproduce ☕️

I am happy to setup a test login account on my Mindbody OAuth client and share the minimal reproduction if needed. For now I'm hoping for some guidance on how to intercept and mutate the Mindbody callback before the callback error gets thrown.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by panva** [comment]
https://en.wikipedia.org/wiki/URI_fragment#:~:text=the%20agent%20sends%20the%20URI%20to%20the%20server%2C%20but%20does%20not%20send%20the%20fragment

Change your `response_type` to just `code`.

**Reply by blimpmason** [reply]
Unfortunately Mindbody returns a fragment regardless of the `response_type`.

I'm thinking the best solution is to setup a client side callback route that captures the fragment and passes to the api callback. Unfortunately I can't get that to work either, though again I think the limitation is on the Mindbody side so may not be worth discussing here unless others find it useful.

**Reply by panva** [reply]
Try passing both response_mode=query with response_type=code then. Either way it's non conform but give it a shot.
